src/main/java/uk/nhs/tis/trainee/notifications/service/NotificationService.java

```diff
@@ -219,17 +223,22 @@ public Map<String, String> executeNow(String jobKey, JobDataMap jobDetails) {
 
         try {
 
+          // we 'send' (i.e. log) the email even if it is an unnecessary reminder, so that the
+          // History is correctly updated.
           emailService.sendMessage(personId, userAccountDetails.email(), notificationType,
-              templateVersion.get(), jobDetails.getWrappedMap(), tisReferenceInfo,
-              !shouldActuallySendEmail(notificationType, personId, tisReferenceInfo.id()));
+              templateVersion.get(), jobDetails.getWrappedMap(),
+              notificationSummary.tisReferenceInfo(),
+              !shouldActuallySendEmail(
+                  notificationType, personId, notificationSummary.tisReferenceInfo().id(),
+                  notificationSummary.unnecessaryReminder()));
```

Comment:
This is quite ugly. Even for reminders that shouldn't be sent, because all actions are complete, we need to ensure that any other previously scheduled notifications are deleted from the History. This is currently all handled in the EmailService, so we 'log' the notification instead of sending it, to allow that processing of the History to take place.

Reply:
Done away with, by rather making overdue reminders 'excluded' and hence not logged or otherwise handled by the EmailService.